src/microbe_stage/PatchRegion.cs

```diff
@@ -0,0 +1,304 @@
+            }
+
+            if (RegionType is "vents" or "underwater_cave")
+            {
+                Patches[0].ScreenCoordinates = new Vector2(ScreenCoordinates.x + regionMargin,
+                    ScreenCoordinates.y + regionMargin);
+
+                // Caves or vents are the same depth as the adjacent patch
+                Patches[0].Depth[0] = Patches[0].Adjacent.First().Depth[0];
+                Patches[0].Depth[1] = Patches[0].Adjacent.First().Depth[1];
+            }
+        }
+    }
+
+    public void BuildRegion()
+    {
+        // Region size configuration
+        Width += 64f + 2 * PatchMargin + RegionLineWidth;
+
+        if (RegionType == "continent")
+        {
+            Height = 64f + 2 * PatchMargin + RegionLineWidth;
+            if (Patches.Count > 1)
+                Width += 64f + PatchMargin;
+
+            if (Patches.Count > 2)
+                Height = 3 * PatchMargin + 2 * 64f + RegionLineWidth;
+        }
+
+        if (RegionType is "ocean" or "sea")
+        {
+            Height += 64f * Patches.Count + (Patches.Count + 1) * PatchMargin + RegionLineWidth;
+        }
+
+        if (RegionType == "vents")
+        {
+            Height = Width = 64 + 2 * PatchMargin + RegionLineWidth;
+
+            var adjacent = Adjacent.First();
+            ScreenCoordinates = adjacent.ScreenCoordinates + new Vector2(0, adjacent.Height) + new Vector2(0, 20);
+        }
+
+        if (RegionType == "underwater_cave")
+        {
+            Height = Width = 64 + 2 * PatchMargin + RegionLineWidth;
+
+            var adjacent = Adjacent.First();
+            var adjacentPatch = Patches[0].Adjacent.First();
+            if (adjacent.RegionType is "sea" or "ocean")
+            {
+                ScreenCoordinates = adjacent.ScreenCoordinates +
+                    new Vector2(adjacent.Width, adjacent.Patches.IndexOf(adjacentPatch) *
+                        (64f + PatchMargin)) + new Vector2(20, 0);
+            }
+
+            if (adjacent.RegionType == "continent")
+            {
+                var leftSide = adjacent.ScreenCoordinates - new Vector2(Width, 0) - new Vector2(20, 0);
+                var rightSide = adjacent.ScreenCoordinates + new Vector2(adjacent.Width, 0) + new Vector2(20, 0);
+                var leftDist = (adjacentPatch.ScreenCoordinates - leftSide).Length();
+                var rightDist = (adjacentPatch.ScreenCoordinates - rightSide).Length();
+
+                if (leftDist < rightDist)
+                {
+                    ScreenCoordinates = leftSide;
+                }
+                else
+                {
+                    ScreenCoordinates = rightSide;
+                }
+
+                ScreenCoordinates = new Vector2(ScreenCoordinates.x,
+                    adjacentPatch.ScreenCoordinates.y - PatchMargin - RegionLineWidth);
+            }
+        }
+    }
+
+    public void ConnectPatchesBetweenRegions(Random random)
+    {
+        if (RegionType is "ocean" or "sea")
```

Comment:
I'll comment this here, which is not probably the best but it seems like there's a **ton** of comparing region type with strings, to me this strongly smells like the region type should be an enum.